projects/components/src/table/data/table-cdk-data-source.ts

```diff
@@ -174,7 +184,12 @@ export class TableCdkDataSource implements DataSource<TableRow> {
       this.filtersProvider.queryProperties$,
       this.columnStateChangeProvider.columnState$,
       this.rowStateChangeProvider.rowState$,
-    ]).pipe(map(values => this.detectRowStateChanges(...values)));
+    ]).pipe(
+      switchMap(values =>
+        combineLatest([of(values), this.tableDataSourceProvider.data?.secondaryDataTriggers$ ?? of(undefined)]),
```

Comment:
Since we are using `combineLatest`, in terms of emitting the loading state value, it will have affect the same way as 2 subscriptions, one to the existing change observable and one to `secondaryDataTriggers$`.

Correct me if I am wrong.